indra/newview/llpanellandmarks.cpp

```diff
@@ -1047,6 +1047,85 @@ bool LLLandmarksPanel::handleDragAndDropToTrash(BOOL drop, EDragAndDropType carg
 	return true;
 }
 
+class LLLandmarkCopiedCallback : public LLInventoryCallback
+{
+public:
+    LLLandmarkCopiedCallback() {}
+
+    virtual void fire(const LLUUID& inv_item)
+    {
+        LLViewerInventoryItem* item = gInventory.getItem(inv_item);
+
+        if (item)
+        {
+            item->setComplete(TRUE);
+            item->updateServer(FALSE);
+
+            gInventory.updateItem(item);
+            gInventory.notifyObservers();
+        }
+
+        LLView::getWindow()->setCursor(UI_CURSOR_ARROW);
+    }
+};
+
+// virtual
+BOOL LLLandmarksPanel::handleDragAndDrop(S32 x, S32 y, MASK mask, BOOL drop, EDragAndDropType cargo_type,
+    void* cargo_data, EAcceptance* accept, std::string& tooltip_msg)
+{
+    *accept = ACCEPT_NO;
+
+    if (cargo_type != DAD_LANDMARK || !cargo_data)
+        return FALSE;
+    LLToolDragAndDrop* tool_dad = LLToolDragAndDrop::getInstance();
+    if (tool_dad->getSource() != LLToolDragAndDrop::SOURCE_LIBRARY)
+        return FALSE;
+    if (tool_dad->getCargoCount() != 1)
```

Comment:
It's a folder, it's valid to drop multiple notecards from inventory, and make sure this won;t cause a regressing where you can no longer drop items from inventory or library.